samples/stored-procedures/clear.js

```diff
@@ -0,0 +1,121 @@
+   */
+  function deleteSessions(sessions) {
+    if (sessions.length > 0) {
+
+      // attempt to delete the first document in the array
+      const accepted = __.deleteDocument(sessions[0]._self, function handler(err) {
+        if (err) throw err;
+
+        responseBody.deleted++;   // increment deleted counter
+        sessions.shift();         // remove document from array
+        deleteSessions(sessions); // delete the next doc
+
+      });
+
+      // if the delete request was not accepted due to timeout, return the {@link responseBody} with a continuation
+      if (!accepted) response.setBody(responseBody);
+
+    } else {
+
+      // if there are no more documents to delete, try getting more
+      getSessions();
+
+    }
+  }
+
+  /**
+   * Filters for session documents based on the provided filter key and value ({@link filterOn}, {@link filterValue}), and immediately begins deleting them as results are returned
+   * @function
+   * @param {String} [continuationToken]   A continuation token, if one was received from a previous request
+   */
+  function getSessions(continuationToken) {
+
+    /**
+     * The filter function that returns a doc only if it has the given filter {@link filterOn} and {@link filterValue}
+     * @function
+     * @param  {Object} doc  The DocumentDB document to test against
+     * @return {Boolean}     Whether the document has the given filter key and value
+     */
+    const filter = function filter(doc) {
+      return doc[filterOn] === filterValue;
+    };
+
+    /**
+     * Handler for the filter request
+     * @function
+     * @param  {Object} err                 The error object, if any was thrown
+     * @param  {Number} err.number          The error code
+     * @param  {String} err.body            The body of the error message
+     * @param  {Array} sessions             The retrieved sessions
+     * @param  {Object} info                Info about the request, including a continuation token
+     * @param  {String} info.continuation   The continuation token, if any was passed
+     * @return {responseBody}
+     */
+    const handler = function handler(err, sessions, info) {
+      if (err) throw err;
+
+      if (sessions.length > 0) {
+
+        // if sessions were found, begin deleting them immediately (prioritizes deletion over searching)
+        deleteSessions(sessions);
+
+      } else if (info.continuation) {
+
+        // if the filter came back empty but with a continuation token, get the next set of results
+        getSessions(info.continuation);
+
+      } else {
+
+        // if there are no more documents and no continuation token, return the {@link responseBody} without a continuation
+        responseBody.continuation = false;
+        response.setBody(responseBody);
+
+      }
+
+    };
+
+    // filter the collection for sessions using the filter function
+    const accepted = __.filter(filter, { continuation: continuationToken }, handler);

```

Comment:
Perf: the predicate function needs to be inlined in order to take advantage of index (otherwise it will be full scan).